processor/src/main/java/org/mapstruct/ap/internal/model/source/SourceReference.java

```diff
@@ -320,6 +320,17 @@ public SourceReference copyForInheritanceTo(SourceMethod method) {
         return new SourceReference( replacement, propertyEntries, isValid );
     }
 
+    public SourceReference pop() {
+        if ( propertyEntries.size() > 1 ) {
```

Comment:
hey.. is this also done now for the TargetProperties popping? I wanted to do that, but did not get to it.